core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -914,17 +914,20 @@ private boolean isPrepared() {
      * @param hintsMap all parameters influencing the weighting. E.g. parameters coming via
      *                 GHRequest.getHints or directly via "&amp;api.xy=" from the URL of the web UI
      * @param encoder  the required vehicle
+     * @param graph The Graph, prefereably the QueryGraph is passed to a Weighting if neccessary
      * @return the weighting to be used for route calculation
      * @see HintsMap
      */
-    public Weighting createWeighting(HintsMap hintsMap, FlagEncoder encoder) {
+    public Weighting createWeighting(HintsMap hintsMap, FlagEncoder encoder, Graph graph) {
         String weighting = hintsMap.getWeighting().toLowerCase();
 
         if (encoder.supports(GenericWeighting.class)) {
             DataFlagEncoder dataEncoder = (DataFlagEncoder) encoder;
             ConfigMap cMap = dataEncoder.readStringMap(hintsMap);
-            cMap = setupBlocking(cMap, hintsMap, dataEncoder);
-            return new GenericWeighting(dataEncoder, cMap);
+            cMap = setupBlocking(cMap, hintsMap, dataEncoder, graph);
+            GenericWeighting genericWeighting =  new GenericWeighting(dataEncoder, cMap);
+            genericWeighting.setGraph(graph);
```

Comment:
hmmh, maybe remove it from the interface and just have the graph in the constructor?